src/modules/esl-utils/dom/rect.ts

```diff
@@ -82,11 +82,13 @@ export class Rect {
    * It increases the size of the Rect by moving each point on the edge of the Rect to a certain distance further away from the center of the Rect.
    * @param increment - distance to grow in pixels
    */
-  public grow(increment: number): Rect {
-    this.y -= increment;
-    this.x -= increment;
-    this.height += 2 * increment;
-    this.width += 2 * increment;
+  public grow(increment: number): Rect;
+  public grow(incrementX: number, incrementY: number): Rect;
```

Comment:
Ideally, please add a description for all of oveloads of the method. 
Because TS separates them as independent parts of final API.

```typescript
  /** Hey I'm method with no arg, I do the following thing... */
  public method(): void;
   /** Hey I'm method with one arg, I may behave diferently... */
  public method(arg1: type): void;
  /** Hey I'm method with one arg by different type, I have a differences with previous implementations... */
  public method(arg1: type): void;
  public method(arg?: any): void { // No TSDoc for exact implementation as it hidden from outsideworled
     // Hangling of all the cases
  }
```